File: 3. lengthOfLongestSubstring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        //Set<Character> set = new HashSet<>();
        int point_a = 0;
        int tem = -1;
        int res=0;
        int[] a = new int[128];
        Arrays.fill(a,-1);
        while(point_a < s.length()){
            
            tem=Math.max(tem,a[s.charAt(point_a)]);
            a[s.charAt(point_a)] = point_a;
            res = Math.max(res,point_a-tem);
            point_a++;
            
        }
        return res;
    }
}

File: 3. lengthOfLongestSubstring.c
int lengthOfLongestSubstring(char * s){
    int res=0, pivot = -1;
    int map[128];
    
    for(int i=0;i<128;i++) map[i]=-1;
    
    for(int i = 0;i<strlen(s);i++){
        pivot=(pivot > map[s[i]]?pivot:map[s[i]]);
        map[s[i]] = i;    
        res = (res > (i- pivot))?res:(i- pivot);
    }
    return res;
}
